tasks/create-internal-request/create-internal-request.yaml

```diff
@@ -53,6 +53,46 @@ spec:
     - name: input
       description: Workspace to store the params and responses for the internalRequest
   steps:
+    - name: merge-internal-request-parameters
+      image:
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      script: |
+          #!/usr/bin/env bash
+          DATA_FILE="$(workspaces.input.path)/$(params.dataPath)"
+          if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+          fi
+
+          IR_PARAMS_FILE="$(workspaces.input.path)/$(params.subdirectory)/ir-params-$$.txt"
+          cat > ${IR_PARAMS_FILE} <<JSON
+          $(params.internalRequestParams)
+          JSON
+
+          # to keep the compatibility the internalrequest parameters must be merged
+          # with the data json
+          JSON_PARAMS="$(jq -c '.' ${IR_PARAMS_FILE})"
+          INDEX=$(jq '.|length' ${IR_PARAMS_FILE})
+          jq '.fbc' $(workspaces.input.path)/$(params.dataPath) |yq -oy -P |while read PARAM_VALUE; do
+            read -r PARAM VALUE <<< "${PARAM_VALUE}"
+
+            PARAM="${PARAM%*:}"
+            # parameters from the task should replace the ones from the data.json
+            if [[ -n $(jq ".[] | select(.name==\"${PARAM}\")" $IR_PARAMS_FILE) ]]; then
+              continue
+            fi
+
+            # xargs is used to remove the extra ""
+            JSON_PARAMS=$(jq ".[${i}] += {name: \"${PARAM:}\", value: \"$(echo ${VALUE}| xargs)\"}" \
+              <<< "${JSON_PARAMS}")
+            #
+            # bash can not remember variables changed inside loops so it is required to keep modifying
```

Comment:
This statement is not true:
```sh
PARAM="a"; for i in 1 2 3; do echo $PARAM; PARAM=${PARAM}a; done
a
aa
aaa
```

The issue here is that in this case you use pipes to execute the loop: `jq | yq | while` and so all of this is executed in a sub shell. But this won't be an issue if the loop is modified to iterate over indexes as I suggest above.

Reply:
Yep. Let me know if you need help with this. I'd be happy to play around with it :)

Reply:
it is changed already. Please take a look.

Reply:
Yep, thanks.